src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,126 @@
+    private static final String QUERY_CREATE_BOOK
+            = "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String QUERY_DELETE_BOOK_BY_ID = "DELETE FROM books WHERE id = ?";
+    private static final String QUERY_FIND_ALL_BOOKS = "SELECT * FROM books";
+    private static final String QUERY_FIND_BOOK_BY_ID = "SELECT * FROM books WHERE id = ?";
+    private static final String QUERY_UPDATE_BOOK
+            = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection
+                        .prepareStatement(QUERY_CREATE_BOOK, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(PARAMETER_INDEX_ONE, book.getTitle());
+            statement.setBigDecimal(PARAMETER_INDEX_TWO, book.getPrice());
+            if (statement.executeUpdate() < EDGE_VALID_NUMBER_AFFECTED_ROWS) {
+                throw new RuntimeException(EXCEPTION_NO_INSERT_BOOK);
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(COLUMN_INDEX, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(EXCEPTION_CAN_NOT_ADD_BOOK + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(QUERY_FIND_BOOK_BY_ID)) {
+            statement.setLong(PARAMETER_INDEX_ONE, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                String title = resultSet.getString(COLUMN_LABEL_TITLE);
+                BigDecimal price = resultSet.getObject(COLUMN_LABEL_PRICE, BigDecimal.class);
+                return Optional.of(new Book(id, title, price));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(EXCEPTION_NOT_FIND_BOOK_BY_ID + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(QUERY_FIND_ALL_BOOKS)) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Book> bookList = new ArrayList<>();
+            while (resultSet.next()) {
+                String title = resultSet.getString(COLUMN_LABEL_TITLE);
+                BigDecimal price = resultSet.getObject(COLUMN_LABEL_PRICE, BigDecimal.class);
+                Long id = resultSet.getObject(COLUMN_LABEL_ID, Long.class);
```

Comment:
Let's create a separate method that will map result set properties to the book object.